ci/docs/hypershift_setup.md

```diff
@@ -0,0 +1,79 @@
+# Hypershift
+
+HyperShift is middleware for hosting OpenShift control planes at scale that solves for cost and time to provision, as well as portability across cloud service providers with strong separation of concerns between management and workloads. Clusters are fully compliant OpenShift Container Platform (OCP) clusters and are compatible with standard OCP and Kubernetes toolchains.
+
+# Steps to setup Hypershift on ROSA
+1. Login to the ROSA cluster created using [ROSA Provisioning](https://github.com/openshift-pipelines/pipeline-service/tree/main/ci/docs/rosa_cluster_provision.md) documentation.
+```bash
+oc login <API_URL>  -u <USERNAME> -p <PASSWORD>
+```
+2. Create MultiClusterEngine to enable Hypershift
+```yaml
+apiVersion: multicluster.openshift.io/v1
+kind: MultiClusterEngine
+metadata:
+  name: multiclusterengine
+spec:
+  overrides:
+    components:
+    - name: hypershift-preview
+      enabled: true
+```
+3. Create ManagedCluster to use this OCP cluster as management cluster for Hypershift. Visit [documentation](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html-single/multicluster_engine/index#hosted-control-planes-configure) for detailed information.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  labels:
+    local-cluster: "true"
+    cloud: auto-detect
+    vendor: auto-detect
+  name: local-cluster
+spec:
+  hubAcceptsClient: true
+  leaseDurationSeconds: 60
+```
+4. Create an S3 bucket for Hypershift Operator with public-read
+```bash
+$BUCKET_NAME=plnsvcci-oidc
```

Comment:
Instead of documenting the commands, maybe this can be put in a script (e.g. `ci/hack/hypershift_setup.sh`?)

The doc would become quite small and easy to follow, as it would explain what Hypershift, list the requirement(s) and then link to the script.

Reply:
Agree with you +1